src/Confluent.Kafka.Avro/AvroSerializer.cs

```diff
@@ -40,118 +36,25 @@ namespace Confluent.Kafka.Serialization
     /// </remarks>
     public class AvroSerializer<T> : ISerializer<T>
     {
-        private SpecificWriter<T> avroWriter;
-
-        private HashSet<string> topicsRegistered = new HashSet<string>();
+        private bool autoRegisterSchema = true;
+        private int initialBufferSize = DefaultInitialBufferSize;
+        private bool isKeySerializer;
+        private ISchemaRegistryClient schemaRegistryClient;
 
         private bool disposeClientOnDispose;
 
-        private string writerSchemaString { get; set; }
+        private IAvroSerializerImpl<T> serializerImpl;
 
         private const string InitialBufferSizePropertyName = "avro.serializer.buffer.bytes";
 
         private const string AutoRegisterSchemaPropertyName = "avro.serializer.auto.register.schemas";
 
-        /// <summary>
-        ///     The SchemaId corresponding to type <see cref="T"/>, or null if 
-        ///     the serializer has yet to be invoked.
-        /// </summary>
-        public int? SchemaId { get; private set; } = null;
```

Comment:
removed this from the public api, because in the general case, the serializer manages more than one schema id. note that this has not yet been included in a release version.

Reply:
(in particular this does not fit with using GenericRecord)